bin/tx-prover/src/proxy/mod.rs

```diff
@@ -40,191 +47,198 @@ impl LoadBalancer {
         }
     }
 
-    /// Create a 429 response for too many requests
-    pub async fn create_too_many_requests_response(
-        session: &mut Session,
-        max_request_per_second: isize,
-    ) -> Result<bool> {
-        // Rate limited, return 429
-        let mut header = ResponseHeader::build(429, None)?;
-        header.insert_header("X-Rate-Limit-Limit", max_request_per_second.to_string())?;
-        header.insert_header("X-Rate-Limit-Remaining", "0")?;
-        header.insert_header("X-Rate-Limit-Reset", "1")?;
-        session.set_keepalive(None);
-        session.write_response_header(Box::new(header), true).await?;
-        Ok(true)
+    /// Gets an available worker and removes it from the list of available workers.
+    ///
+    /// If no worker is available, it will return None.
+    pub async fn get_available_worker(&self) -> Option<Backend> {
```

Comment:
nit: could we rename this to `pop_available_worker` just to make it clear that it's a get+remove

Reply:
much better